cmd/fs_mgmt/port/zephyr/src/zephyr_fs_mgmt.c

```diff
@@ -110,24 +111,47 @@ int
 fs_mgmt_impl_write(const char *path, size_t offset, const void *data,
                    size_t len)
 {
-    struct fs_file_t file;
+    static struct fs_file_t file;
+    static char *previous_path = NULL;
```

Comment:
There's one specific flow that I'm worried about here when using these statics.

1) User uploads a file with mcumgr
2) User deletes the file 
3) User uploads a _different_ file to the same path. 

In this flow would it be the case that `file` wouldn't point to the right place?

Reply:
Good point. Treating `offset == 0` as a re-initialization case covers that case, so I added to the re-initialization conditions in 8de7744.